python/plugins/processing/algs/qgis/ui/HeatmapWidgets.py

```diff
@@ -127,8 +131,9 @@ def rowsChanged(self):
 
     def columnsChanged(self):
         cols = self.mColumnsSpinBox.value()
-        if cols < 2:
+        if cols < 2 or self.raster_bounds.isNull():
             return
+
         cell_size = self.raster_bounds.width() / (cols - 1)
```

Comment:
Should a return be added here if cell_size = 0 ( would be if the raster_bounds are an vertical line or a point )